persistence/rest/src/main/java/org/infinispan/persistence/rest/configuration/RestStoreConfigurationParser.java

```diff
@@ -168,6 +175,14 @@ private void parseRestStoreAttributes(XMLExtendedStreamReader reader, RestStoreC
             }
          }
       }
+
+      String cacheName = reader.getSchema().since(10, 1) ? cacheNameValue : getCacheNameFromLegacy(appendCacheName, path);
```

Comment:
I think the `since(10, 1)` check needs to go in the switch cases for the legacy attributes.

Reply:
Not sure I get it. It will also try to read all the attributes (legacy and non legacy) and then afterwards, it will try to extract the cache name differently if it's older than 10.1

Reply:
I also added a test to parse 9.4.xml files

Reply:
@gustavonalle Sorry, I thought I was clearer on Zulip: I meant if the schema is >= 10.1, don't parse the legacy attributes and throw an exception instead, like we do in the main parser:

https://github.com/infinispan/infinispan/blob/9490d0d006fa6941a02e9a2731937dfb8306ec52/core/src/main/java/org/infinispan/configuration/parsing/Parser.java#L1148

It's not a deal breaker, but I like it when all the modules use the same conventions.

Reply:
ok, I will fix it

Reply:
@danberindei Fixed. I won't mark the conversation resolved this time :smile: